ta/os_test/ta_arm_pauth.c

```diff
@@ -32,11 +37,8 @@ TEE_Result ta_entry_pauth_test_nop(void)
 		return TEE_ERROR_NOT_SUPPORTED;
 
 	/* Check if PAC instruction generates the authentication code */
-	for (int i = 0; i < 10; i++) {
-		asm volatile("paciza %0 " : "+r" (ptr));
-		pac |= ptr & VA_MASK;
-		ptr = i;
-	}
+	for (int i = 0; i < 10; i++)
+		pac |= sign_using_keyia(i) & VA_MASK;
 
 	if (implemented && pac)
```

Comment:
Should `pac != 0 ` be checked for each of the _paciza_ instructions in the loop above?

Reply:
I intentionally didn't add this check. Here we are assuming VA_MASK with 48 bits of VA. Since the VA MASK used over here is not exact, there are minimal chances that authentication code generated may be 0 for the mask used. Hence I repeated it over a range of numbers and ored the result. 

The main purpose of this test case is basically to ensure that pac is enabled and these instructions are not treated as nops.